File: examples/python/multiple_networks.py
from webweb import Web

# define the node names and a hunger metadata attribute to appear in all networks
web = Web(
    display={
        'nodes' : {
            0 : {
                'name' : 'dane',
                'hunger' : 4,
            },
            1 : {
                'name' : 'sebastian',
                'hunger' : 9,
            },
            2 : {
                'name' : 'manny',
                'hunger' : 2,
            },
            3 : {
                'name' : 'brock',
                'hunger' : 4,
            },
            4 : {
                'name' : 'ted',
                'hunger' : 12.1,
            },
            5 : {
                'name' : 'donnie',
                'hunger' : 5,
            },
        }
    },
)

# add a 'snake' network
web.networks.snake(
    adjacency=[[0, 1], [1, 2], [2, 3], [3, 4], [4, 5]],
    nodes={
        0 : {
            'isHead' : False,
        },
        1 : {
            'isHead' : False,
        },
        2 : {
            'isHead' : False,
        },
        3 : {
            'isHead' : False,
        },
        4 : {
            'isHead' : False,
        },
        5 : {
            'isHead' : True,
        },
    },
)

# add a 'starfish' network
web.networks.starfish(
    adjacency=[[0, 1], [0, 2], [0, 3], [0, 4], [0, 5]],
    nodes={
        0 : {
            'texture' : 'gooey',
            'power' : 1,
        },
        1 : {
            'texture' : 'fishy',
            'power' : 3,
        },
        2 : {
            'texture' : 'chewy',
            'power' : 3.8,
        },
        3 : {
            'texture' : 'crunchy',
            'power' : 0.2,
        },
        4 : {
            'texture' : 'chewy',
            'power' : 1,
        },
        5 : {
            'texture' : 'gooey',
            'power' : 3.1415,
        },
    }
)

# display the `starfish` network first
web.display.networkName = 'starfish'

# we'll compute node color by the `texture` attribute
web.display.colorBy = 'texture'

# we'll compute node size by the `isHead` attribute
web.display.sizeBy = 'isHead'

web.draw()

File: examples/matlab/multiple_networks.m
% set some node names and a metadata attribute (hunger) to appear for all
% of the networks that we will display. Since these metadata attributes
% apply globally, we put them under ww.display
ww.display.metadata.names = {'dane','sebastian','manny','brock','ted','donnie'};
ww.display.metadata.hunger = [4,9,2,4,12.1,5];

% define a network called snake
ww.networks.snake.edgeList = [1,2; 2,3; 3,4; 4,5; 5,6];
ww.networks.snake.metadata.isHead.values = [0,0,0,0,0,1];

% define another called starfish
ww.networks.starfish.edgeList = [1,2; 1,3; 1,4; 1,5; 1,6];
ww.networks.starfish.metadata.texture.values={'gooey','fishy','chewy',...
    'crunchy','chewy','gooey'};
ww.networks.starfish.metadata.power.values={1,3,3.8,0.2,1,3.1415};

% BONUS force the starfish network to display first by default
ww.display.networkName = 'starfish';
% BONUS force node color to texture metadata
ww.display.colorBy = 'texture';
% BONUS force node size to isHead metadata
ww.display.sizeBy = 'hunger';

webweb(ww)
